### PR TITLE
[MVC] Move samples to GenericHost

### PR DESCRIPTION
**Pull request by Kahbazi**
Part of #20964

### CODE REVIEW

File: src/Mvc/test/WebSites/BasicWebSite/Program.cs [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
     public class Program
     {
-        public static void Main(string[] args) => CreateWebHostBuilder(args).Build().Run();
+        public static Task Main(string[] args)
+        {
+            var host = CreateHostBuilder(args)
+                .Build();
 
-        // Do not change. This is the pattern our test infrastructure uses to initialize a IWebHostBuilder from
-        // a users app.
```
Comment (Kahbazi) [inline src/Mvc/test/WebSites/BasicWebSite/Program.cs:15]: Is this comment still valid?

**Comment by Kahbazi** [comment]
Should I remove this test, or create a `WebHostBuilder` just for this test?

https://github.com/dotnet/aspnetcore/blob/4fde84a7ad48c8265fe868650351c857bcf1bfc2/src/Mvc/test/Mvc.FunctionalTests/TestingInfrastructureInheritanceTests.cs#L19

**Comment by BrennanConroy** [comment]
@javiercn Could you please take a look and answer the questions above

File: src/Mvc/test/WebSites/BasicWebSite/Program.cs (hunk 1 above)
Comment (javiercn) [reply src/Mvc/test/WebSites/BasicWebSite/Program.cs:15]: We still support both patterns (WebHostBuilder hasn't gone away) so we should keep coverage for this.

Comment (javiercn) [reply src/Mvc/test/WebSites/BasicWebSite/Program.cs:15]: So even though it's obsolete we still need to make sure it keeps working.

**Comment by pranavkm** [comment]
@Kahbazi do you expect to continue working on this PR?

**Comment by Kahbazi** [comment]
@pranavkm sorry for the late reply. Is it possible for someone else to continue on this PR?

**Comment by javiercn** [comment]
@Kahbazi We'll take over it

**Comment by javiercn** [comment]
We are not taking this change as it still needs some work on our part to make it work and this area is not a priority for us at the moment. We might revisit this in the future and reopen the PR as a result.

**Pull request closed without merging**